documentation/specs/rar-as-service.md

```diff
@@ -0,0 +1,217 @@
+
+    public string TargetFrameworkMoniker { get { throw null; } set { } }
+
+    public string TargetFrameworkMonikerDisplayName { get { throw null; } set { } }
+
+    public string[] TargetFrameworkSubsets { get { throw null; } set { } }
+
+    public string TargetFrameworkVersion { get { throw null; } set { } }
+
+    public string TargetProcessorArchitecture { get { throw null; } set { } }
+
+    public bool UnresolveFrameworkAssembliesFromHigherFrameworks { get { throw null; } set { } }
+
+    public string WarnOrErrorOnTargetArchitectureMismatch { get { throw null; } set { } }
+
+    public Microsoft.Build.Framework.LoggerVerbosity Verbosity { get { throw null; } set { } }
+}
+```
+
+### RAR Output
+
+Output of RAR node has to also reflect RAR task.
+```csharp
+public sealed partial class ResolveAssemblyReferenceOutput
+{
+    public ResolveAssemblyReferenceOutput() { }
+
+    public Microsoft.Build.Framework.ITaskItem[] CopyLocalFiles { get { throw null; } }
+
+    public string DependsOnNETStandard { get { throw null; } }
+
+    public string DependsOnSystemRuntime { get { throw null; } }
+
+    public Microsoft.Build.Framework.ITaskItem[] FilesWritten { get { throw null; } }
+
+    public Microsoft.Build.Framework.ITaskItem[] RelatedFiles { get { throw null; } }
+
+    public Microsoft.Build.Framework.ITaskItem[] ResolvedDependencyFiles { get { throw null; } }
+
+    public Microsoft.Build.Framework.ITaskItem[] ResolvedFiles { get { throw null; } }
+
+    public Microsoft.Build.Framework.ITaskItem[] SatelliteFiles { get { throw null; } }
+
+    public Microsoft.Build.Framework.ITaskItem[] ScatterFiles { get { throw null; } }
+
+    public Microsoft.Build.Framework.ITaskItem[] SerializationAssemblyFiles { get { throw null; } }
+
+    public Microsoft.Build.Framework.ITaskItem[] SuggestedRedirects { get { throw null; }
+}
+```
+### Logging response
+
+Node also has to return logged data, preferably in stream which will be transferred during the RAR task execution to the client node. This should by done by using IProgress\<T> support in StreamJsonRPC. There is also support for async enumerable, but they must be pulled and are not pushed to the client (comparison can be found [here](https://github.com/microsoft/vs-streamjsonrpc/blob/master/doc/asyncenumerable.md#comparison-with-iprogresst)).
+
+### Concurrency
+
+One of the biggest challenges with RAR as service, is to make execution and caching of RAR task thread-safe, since in most cases there will be multiple clients request data from it at once.
+
+In the first phase of implementation the concurrency will be solved by serializing requests, but this will be replaced by optimal variant with no serializing.
+
+# Non-Goals
+
+- File watchers: using them would decrease required IO operations when checking disc changes
+- Aggressive precomputation of results
+- Improved caching of requests
+- Providing verbosity to RAR task
```

Comment:
Thank you for explicitly listing it as a non-goal. Should the text about logging under "RAR Input", as well as the `Verbosity` property in `ResolveAssemblyReferenceInput` be deleted, then?